src/main/webapp/index.html

```diff
@@ -44,37 +46,62 @@ <h2 class="panel-row__header u-align-with-first-panel u-text-shadow">Services (a
+          <img class="panel__toggle-icon--unselected" alt="settings-icon" src="assets/settings-unselected.png" onclick="displayGmailSettings()">
+        </div>
+        <div class="panel__content-entry u-remove-top-margin gmail-content">
+          <a class="panel__content-entry-link" target="_blank" href="https://www.gmail.com">
+            <img class="panel__logo" alt="mail-logo" src="assets/mail.png">
+          </a>
+          <h3 class="panel__section-heading">Mail</h3>
+        </div>
+        <p class="panel__section-subheading gmail-content">From the last <span id="gmail-n-days"></span> days:
+        <div class="panel__content gmail-content">
+          <div class="panel__content-entry">
+            <div class="panel__rounded-number u-background-mail">
+              <p class="panel__text-data-point u-text-white" id="gmail-unread-emails-days">
             </div>
-            <div class="panel__content-entry">
-              <img class="panel__content-icon" alt="clock-icon" src="assets/clock.png">
-              <p class="panel__text-large"><span id="gmailUnreadEmailsHours"></span> from the last <span id="gmailMHours"></span> hours
+            <p class="panel__text-data-point">Unread Emails:
+          </div>
+          <div class="panel__content-entry">
+            <img class="panel__content-icon" alt="clock-icon" src="assets/clock.png">
+            <p class="panel__text-large"><span id="gmail-unread-emails-hours"></span> from the last <span id="gmail-m-hours"></span> hours
+          </div>
+          <div class="panel__content-entry">
+            <img class="panel__content-icon" alt="flag-icon" src="assets/flag.png">
+            <p class="panel__text-large"><span id="gmail-unread-important-emails"></span> marked important
+          </div>
+          <div class="panel__content-entry">
+            <div class="panel__round-letter">
+              <p class="panel__text-data-point u-text-white" id="gmail-sender-initial">
             </div>
-            <div class="panel__content-entry">
-              <img class="panel__content-icon" alt="flag-icon" src="assets/flag.png">
-              <p class="panel__text-large"><span id="gmailUnreadImportantEmails"></span> marked important
+            <p class="panel__text-large"><span id="gmail-sender"></span> sent the most unread emails
+          </div>
+        </div>
+        <!-- Settings section -->
+        <h3 hidden class="panel__section-heading u-remove-top-margin gmail-settings">Settings:</h3>
+        <div hidden class="panel__text-body gmail-settings">
+          <div class="panel__content-entry u-remove-top-margin">
+            <p class="panel__text-large">Search <span class="u-bold-underline" id="gmail-settings-n-days">7</span> days of email
+            <div class="panel__two-buttons-incremental u-right-align">
+              <a class="panel__button-incremental" onclick="incrementElement('gmail-settings-n-days')"><span>+</span></a>
```

Comment:
Is the increment limited to 30? What do you think of making it "infinite" if the user tries to go beyond the limit?

I think it would also be cool if the "+" disappeared when the user reached "infinity" or the limit depending on what you decide (same for "-"). (This only makes sense if it's easy to implement.)

Reply:
I think that we should have a limit for performance reasons. 

Batching should help with this, but I know power users (like me) have a **lot** of email. I think I have roughly 100,000 unread emails in my Gmail. I'm worried that if we remove the cap, the system will not perform well even with the best methods implemented. 

I wanted the cap to be generous, and make semantic sense, so I thought 30 would be a good idea - let me know if you think a different limit would be better (I just picked this arbitrarily).

In terms of adding and removing the +, I can toggle the disabled feature on the button which may work out nicely, but that's going to take some time (~1 hour). If I have some time tomorrow, I'll take a look into it. Good idea 👍

Reply:
I did it!

Reply:
Cool, looks great!

Regarding the cap, I think it would be nice to give power users the "power" to parse all their emails if they wanted, but that's totally minor. I think the +/- fading out you did is pretty cool and wraps up this discussion in a great way!

Reply:
Thanks! 

RESOLVED